### DISCUSSION TITLE
Use externalbrowser authenticator?

### DISCUSSION THREAD

**Discussion by JohnL4** [littleK0i/SnowDDL#77] [Q&A] [answered]
Hey, crew, I did a little research and didn't find anything, but: is it possible to use an externalbrowser authenticator with the CLI tools (snowddl-converter, snowddl)? I have the access I need to a d/b via SSO, but not via the service account whose password I know. Did I miss something obvious?

(P.S. On Windows.)

John.

**Comment by littleK0i** [comment, marked as answer]
I did not consider SSO as viable option for CLI tools. You would normally want to use password or private key for scripts.

However, according to this article, it seems like Snowflake Python connector already implements all the groundwork:
https://community.snowflake.com/s/article/Using-Single-Sign-On-SSO-with-Python-using-Jupyter-Notebook

I'll see if I can add it in next versions.

In the meantime, if you need it urgently, you may overload this method:
https://github.com/littleK0i/SnowDDL/blob/master/snowddl/app/base.py#L385-L415

And add connection option: `"authenticator": "externalbrowser",`

In theory, it should work out-of-the-box.

**Reply by JohnL4** [reply]
Thank you!

I asked our db admin to grant me permission and, to my surprise, he actually did set me up with a new service account, so we're off the hook for now.

I will eventually be scripting this, but, for now, I'm in exploration/POC mode.

John.

**Comment by ghost** [comment]
I'm interested in this feature as well:

> I'll see if I can add it in next versions.

**Reply by littleK0i** [reply]
It is available in the latest version.

Try adding `--authenticator externalbrowser` to CLI options.

If you want SSO caching, try installing with `snowddl[sso-caching]` optional dependency.

I'll update docs when I have a bit more time.

**Comment by JohnL4** [comment]
W00t!

--
Sent from my tablet,  which has a funny keyboard. Makes me sound more curt
and muted than normal.
